crypto/asn1/a_gentm.c

```diff
@@ -163,56 +223,49 @@ ASN1_GENERALIZEDTIME *ASN1_GENERALIZEDTIME_set(ASN1_GENERALIZEDTIME *s,
     return ASN1_GENERALIZEDTIME_adj(s, t, 0, 0);
 }
 
+int ASN1_GENERALIZEDTIME_cmp_time_t(const ASN1_GENERALIZEDTIME *s, time_t t)
```

Comment:
strcmp() is insufficient to compare times when that have not been normalized; comparing a pre-2049 UTCTIME with a post-2050 GENERALIZEDTIME (as required by certificates) will not work.